### ISSUE TITLE
AbstractMaildirDomain implements IAliasableDomain but doesn't advertise that fact.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6218]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)| @tomprince reported|
|-|-|
|Trac ID|trac#6218|
|Type|enhancement|
|Created|2012-12-22 22:26:08Z|
|Branch|https://github.com/twisted/twisted/tree/maildirdomain-iface-6218|

<details><summary>Searchable metadata</summary>

```
trac-id__6218 6218
type__enhancement enhancement
reporter__tom_prince tom.prince
priority__normal normal
milestone__ 
branch__branches_maildirdomain_iface_6218 branches/maildirdomain-iface-6218
branch_author__thijs thijs
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__ 
time__1356215168000000 1356215168000000
changetime__1361799982000000 1361799982000000
version__None None
owner__thijs thijs
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_1"></a>|@thijstriemstra commented|
|-|-|

(In [37216]) Branching to 'maildirdomain-iface-6218'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

(In [37219]) add interface, news file. refs [#6218](https://github.com/twisted/twisted/issues/6218)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra commented|
|-|-|

Forced a [build](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/maildirdomain-iface-6218), up for review.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_4"></a>|@tomprince set owner to @thijstriemstra|
|-|-|

  1. Please add a test (using `verifyClass`) that it does in fact implement that interface.
  1. Implementing `IDomain` is redundant, since `IAliasableDomain` inherits from it.
  1. Similarly, the `IAliasableDomain` annotation on `MaildirDirdbmDomain` is now superflous.

Please commit after addressing the above issues.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra commented|
|-|-|

(In [37247]) address review comments. refs [#6218](https://github.com/twisted/twisted/issues/6218)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra set status to `assigned`|
|-|-|

Replying to [tom.prince](#note_4):
>   1. Please add a test (using `verifyClass`) that it does in fact implement that interface.
>   1. Implementing `IDomain` is redundant, since `IAliasableDomain` inherits from it.
>   1. Similarly, the `IAliasableDomain` annotation on `MaildirDirdbmDomain` is now superflous.
> 
> Please commit after addressing the above issues.

Thanks for the review. Forced a [new build](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/maildirdomain-iface-6218) and will merge when its green.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [37250]) Merge maildirdomain-iface-6218: AbstractMaildirDomain implements IAliasableDomain and is now marked as such.

Author: thijs
Reviewer: tom.prince
Fixes: [#6218](https://github.com/twisted/twisted/issues/6218)

Also updates other interface definitions in maildir with zope.interface's @implementer decorator.
